File: Java/src/se/liu/albhe576/project/Timer.java
package se.liu.albhe576.project;

public class Timer {

    private long lastTick;
    private long serverTicks;
    private boolean running;
    public long getLastTick(){
        return this.lastTick;
    }

    public void reset(){
        this.running = false;
        this.serverTicks = 0;
        this.lastTick = 0;
    }
    public Timer(){
        this.running = false;
        this.serverTicks = 0;
        this.lastTick = 0;
    }
    public void updateTimer(){
        long ticks = System.currentTimeMillis();
        this.lastTick += ticks - this.serverTicks;
        this.serverTicks = ticks;
    }
    public void startTimer(){
        this.running = true;
        this.serverTicks = System.currentTimeMillis();
    }
    public void stopTimer(){
        this.running = false;
    }
}

File: C/src/timer.c
#include "timer.h"
#include <SDL2/SDL_timer.h>

void resetTimer(Timer *timer)
{
  timer->running     = false;
  timer->lastTick    = 0;
  timer->serverTicks = 0;
}
void updateTimer(Timer *timer)
{
  u64 tick = SDL_GetTicks();
  timer->lastTick += tick - timer->serverTicks;
  timer->serverTicks = tick;
}
void startTimer(Timer *timer)
{
  timer->running     = true;
  timer->serverTicks = SDL_GetTicks();
}
void stopTimer(Timer *timer)
{
  timer->running = false;
}
